js/gdm/loginDialog.js

```diff
@@ -628,10 +633,28 @@ const LoginDialog = new Lang.Class({
         this._passwordHintButton.connect('clicked', Lang.bind(this, this._showPasswordHint));
         this._passwordHintButton.visible = false;
 
+        let passwordResetLabel = new St.Label({ text: _("Forgot password?"),
+                                                style_class: 'login-dialog-password-recovery-link' });
+        this._passwordResetButton = new St.Button({ style_class: 'login-dialog-password-recovery-button',
+                                                    button_mask: St.ButtonMask.ONE | St.ButtonMask.THREE,
+                                                    can_focus: true,
+                                                    child: passwordResetLabel,
+                                                    reactive: true,
+                                                    x_align: St.Align.START,
+                                                    x_fill: true });
+        this._promptBox.add(this._passwordResetButton,
+                            { visible: false,
```

Comment:
Yes, clearly; thanks for the catch. I'm a bit surprised this didn't cause any problems, but I guess as part of the BoxLayout it's just ignored, and the button must be hidden by default anyway.